web-ui/src/main/resources/catalog/components/metadataactions/partials/related.html

```diff
@@ -43,14 +43,14 @@ <h3 data-ng-if="::(r.title | gnLocalized: lang).length">{{::(r.title | gnLocaliz
           <div data-ng-switch-when="WMS">
             <p class="text-muted">
               <span data-translate=""
-                    data-translate-values="{url:'{{r.url | gnLocalized: lang}}', layer:'{{(r.title | gnLocalized: lang).replace('\'', '\\\'')}}'}">
+                    data-translate-values="{url:'{{r.url | gnLocalized: lang}}', layer:'{{r.title | gnLocalized: lang | escapeQuote}}'}">
```

Comment:
It seems like this is only a partial fix that only fixes the escaping quotes in json string.  Seems like if the text contained a "\" it could end up having a similar issue where the text breaks the json object.
It is generally not recommended to try perform manual escaping of the json string based on user input when there are already tools that do this work.

Would it not be better to use JSON.stringify?
See second answer from the following url.
https://stackoverflow.com/questions/4031324/javascript-how-escape-quotes-in-a-var-to-pass-data-through-json

Not sure if the following would work?
data-translate-values="JSON.stringify({url:'{{r.url | gnLocalized: lang}}', layer:'{{(r.title | gnLocalized: lang)}}'})">

Reply:
Good point, thanks for your valuable input.

The `data-translate-values` attribute does not expect JSON but an AngularJS expression, hence why single quotes are accepted. This being said, using JSON instead and relying on the native API is a good idea. I don't think your example would work though since the JSON API is not accessible in angular expressions, but maybe something like this:
```
data-translate-values="{url:{{r.url | gnLocalized: lang}}, layer:{{r.title | gnLocalized: lang | toJson}}}"
```

Instead of introducing a `escapeQuote` filter we'd introduce a `toJson` one. I'll give it a shot and update the PR if that works out.

Reply:
Updated the PR, @ianwallen let me know what you think! Thanks

Reply:
@jahow code looks much better - thank you for applying the update.